src/mscorlib/src/System/Environment.cs

```diff
@@ -1338,6 +1394,8 @@ internal static String GetStackTrace(Exception e, bool needFileInfo)
 #if BIT64
                     // 64-bit programs run only on 64-bit
                     return true;
+#elif PLATFORM_UNIX
+                    return IntPtr.Size == 8; // TODO: use uname and match utsname.machine?
```

Comment:
> The same simplifying assumption should apply here

Meaning Is64BitProcess should always equal Is64BitOperatingSystem, both of which should be true iff BIT64 is set, and both of which should be true iff IntPtr.Size == 8?